### DISCUSSION TITLE
VpnHoodClient.Dispose() not working with cellular network in android.

### DISCUSSION THREAD

**Discussion by mohnishsiyodia** [vpnhood/VpnHood#378] [Q&A] [answered]
I have created an Android application using Xamarin and VpnHood. There is a basic login-logout functionality where on logout button clicked I need to disconnect the vpnhoodclient which is handled by VpnHoodClient.Dispose().

Now this method is working, the vpnhoodclient is disposed and a new session starts, when my android device is connected to a wifi whereas when it is on cellular network, my application logs out but the vpn connection is not disposed.

Please refer to the image attached below of the exception.

![VpnHood_Exception](https://github.com/vpnhood/VpnHood/assets/53507901/fc932a3d-f3b6-4271-b89e-ea629f84a2a1)

**Comment by trudyhood** [comment, marked as answer]
Hi, I am not sure you are using the latest version, As ProxyManager.Dispose does not exist anymore in favor of DisposeAsync. make sure to update your source code or nuget to the latest version
